src/core/pyproxy.c

```diff
@@ -676,13 +686,17 @@ _pyproxy_get_buffer(PyObject* ptrobj)
     return NULL;
   }
   Py_buffer view;
+  // PyBUF_RECORDS_RO requires that suboffsets be NULL but otherwise is the most
+  // permissive possible request.
   if (PyObject_GetBuffer(ptrobj, &view, PyBUF_RECORDS_RO) == -1) {
-    PyErr_Clear();
-    return NULL;
+    // Buffer cannot be represented without suboffsets. The bf_getbuffer method
+    // should have set a PyExc_BufferError saying something to this effect.
+    pythonexc2js();
   }
 
   bool success = false;
   if (view.ndim == 0) {
+    // TODO: what should we do here?
```

Comment:
ndim=0 works with numpy,
```py
>>> import numpy as np
>>> x = np.array(0)
>>> x
array(0)
>>> x.ndim
0
>>> x.shape
()
```
and [is defined in the buffer protocol](https://docs.python.org/3/c-api/buffer.html#c.Py_buffer.ndim) so could we return that ?

Reply:
So you think we should return it as a single entry TypedArray?

What about the case when the shape has a 0 in it? You have an opinion about that one?

Reply:
If prod(shapes) == 0, it means there is no data, so I think one would return an buffer with an empty array.

Reply:
Okay I think I fixed it to correctly handle these cases.